components/Modal.js

```diff
@@ -0,0 +1,53 @@
+import { changeData } from '../api/requests.js';
+
+export class Modal {
+  modalWindowContainer = null;
+  bgShadow = null;
+  changedDataObj = {};
+
+  constructor() {}
+
+  renderModal(targetObject, rowId) {
+    this.modalWindowContainer = document.createElement('div');
+    this.modalWindowContainer.classList.add('modal-window-container');
+    this.bgShadow = document.createElement('div');
+    this.bgShadow.classList.add('bg-shadow');
+    document.body.append(this.modalWindowContainer, this.bgShadow);
+    this.bgShadow.addEventListener('click', () => {
+      this.remove();
+    });
+    this.renderInputFields(targetObject, this.changedDataObj);
```

Comment:
We have `changedDataObj` as a property in this class
Not necessary to pass it as an argument